server/journeys/screen/screen/schemas.ts

```diff
@@ -0,0 +1,44 @@
+import z from 'zod'
+import { Request } from 'express'
+import type CsipApiService from '../../../services/csipApi/csipApiService'
+
+const csrfSchema = z.object({
+  _csrf: z.string().optional(),
+})
+
+const saferCustodyOutcomeErrorMsg = 'Select the outcome of Safer Custody screening'
+const REASONS_MSG = 'Enter a description of the reasons for this decision'
+
+const REASONS_TOO_LONG_MSG = 'Description must be 4,000 characters or less'
+
+export const schemaFactory = (csipApiService: CsipApiService) => async (req: Request) => {
+  const saferCustodyOutcomeMap = new Map(
+    (await csipApiService.getReferenceData(req, 'outcome-type')).map(itm => [itm.code, itm]),
+  )
+
+  return z
+    .object({
+      saferCustodyOutcome: z.string({ message: saferCustodyOutcomeErrorMsg }).transform((val, ctx) => {
+        if (!saferCustodyOutcomeMap.has(val)) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: saferCustodyOutcomeErrorMsg,
+          })
+          return z.NEVER
+        }
+        return saferCustodyOutcomeMap.get(val)!
+      }),
+    })
+    .and(
+      csrfSchema.merge(
+        z.object({
+          reasonForDecision: z
+            .string({ message: REASONS_MSG })
```

Comment:
You can just chain `.trim()` on here, then simply use `.min(1, { message: REASONS_MSG })`